File: AngularLovelyjubbly.Data.Sql.Models/PowerRanking.cs
namespace AngularLovelyjubbly.Data.Sql.Models
{
    public partial class PowerRanking
    {
        public PowerRanking()
        {
        }

        public int PowerRankingId { get; set; }
        //foreign key
        public int TournamentId { get; set; }
        //navigation property
        public virtual Tournament Tournament { get; set; }
        //foreign key
        public int WeekId { get; set; }
        //navigation property
        public virtual Week Week { get; set; }
        //foreign key
        public int TeamId { get; set; }
        //navigation property
        public virtual Team Team { get; set; }
        public byte CurrentRanking { get; set; }
        public byte PreviousRanking { get; set; }
    }
}

File: AngularLovelyjubbly/ClientApp/src/app/shared/models/powerranking.ts
import { ITournament } from './tournament';
import { IWeek } from './week';
import { ITeam } from './team';

export interface IPowerRanking {
    powerRankingId: number;
    tournament: ITournament;
    week: IWeek;
    team: ITeam;
    currentRanking: number;
    previousRanking: number;
    statusChangeImage: string;
}

export class PowerRanking implements IPowerRanking {

    constructor(public powerRankingId: number, public tournament: ITournament, public week: IWeek,
        public team: ITeam, public currentRanking: number, public previousRanking: number, public statusChangeImage: string) {
    }
}
